src/components/ebay-list/index.marko

```diff
@@ -0,0 +1,74 @@
+import { processHtmlAttributes } from "../../common/html-attributes";
+$ const {
+    class: inputClass,
+    style,
+    items,
+    ...htmlInput
+} = input;
+
+<div style=style class=["list", inputClass] ...processHtmlAttributes(htmlInput)>
+    <ul>
+        <for|item, i| of=items>
+            $ const {
+                class: itemInputClass,
+                style: itemStyle,
+                as,
+                separator,
+                leading,
+                trailing,
+                ...itemInput
+            } = item;
+            <if(separator)>
+                <hr>
+            </if>
+            <else>
+
+                <li>
+                    <${as || "div"}
+                        class=["list__body", itemInputClass]
+                        style=itemStyle
+                        on-click(
+                            as === "button" && "emit",
```

Comment:
Why only emit this when it's a button? Seems like an arbitrary restriction.

Reply:
This is because when its static or an anchor tag, I don't want to emit an event. 
We shouldnt emit a click event on an anchor tag and on a static div.

Reply:
Why not? `<div on-click("whatever")/>` works, I don't think it makes sense to prevent people from doing that.

Reply:
It does. But in this case we want to emit an event for button click. This is because there is no other way for users to know if something was clicked. 
I don't want to render a static list and emit a click event on each item. The same with an anchor tag. This is because a static list should not be intractable. Same with the switch one. I don't want to emit an event on the switch being toggled because a click event was triggered on the static item.

Reply:
Okay, I see where you're coming from. Maybe this can be revisited/refined when there are other events also being emitted.